kafka-connect-cassandra/src/main/scala/com/datamountaineer/streamreactor/connect/cassandra/sink/CassandraJsonWriter.scala

```diff
@@ -176,6 +190,57 @@ class CassandraJsonWriter(connection: CassandraConnection, settings: CassandraSi
   }
 
 
+  private def delete(records: Seq[SinkRecord]) = {
+    val executor = Executors.newFixedThreadPool(settings.threadPoolSize)
+
+    try {
+      val futures = records map { record =>
+        executor.submit {
+          try {
+            deleteCache match {
+              case Some(d) =>
+                val bindingFields = {
+                  val recordKey = record.key()
+
+                  if (record.keySchema().`type`().isPrimitive) {
+                    logger.trace("key schema is a primitive type, this is easy...")
+                    Seq(record.key())
+                  }
+                  else {
+                    logger.trace("key schema is a STRUCT, dig into the key...")
+                    deleteStructFlds map { f =>
+                      val clazz = recordKey.getClass
```

Comment:
Why not cast to Struct and then get the field by name recordKey.asInstance[Struct].get(fieldName)?

Reply:
Thanks for chiming in with that, because I HATED the solution I came up with (reflection). I will make that change and test.